csharp/ql/lib/semmle/code/csharp/security/dataflow/UrlRedirectQuery.qll

```diff
@@ -184,8 +184,12 @@ private class InterpolationSanitizer extends Sanitizer {
  */
 private class StringFormatSanitizer extends Sanitizer {
   StringFormatSanitizer() {
-    exists(FormatCall c, Expr e | c = this.getExpr() and e = c.getFormatExpr() |
-      e.(StringLiteral).getValue().splitAt("{0}", 0).matches("%?%")
+    exists(FormatCall c, Expr e, int index, string format |
+      c = this.getExpr() and e = c.getFormatExpr()
+    |
+      format = e.(StringLiteral).getValue() and
+      exists(format.regexpFind("\\{[0-9]+\\}", 0, index)) and
+      format.substring(0, index).matches("%?%")
```

Comment:
Alternatively, we could let the _argument_ be the sanitizer (as opposed to the _call_), and then check that the first reference to the argument in the format string is preceded by a `"?"`, but the above is probably sufficient in practice.

Reply:
If we do that then the subsequent uses will also be considered sanitized.

Reply:
Ah, right.